### PR TITLE
DNR properties changes

### PR DESCRIPTION
**Pull request by rebloor**
#### Summary

Adds browser compatibility data in support of [Bug 1894128](https://bugzilla.mozilla.org/show_bug.cgi?id=1894128) 
[DNR] Replace MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES including:

- adding a note that MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES was deprecated in Firefox 128 (I haven't marked the entire property as deprecated given that Safari hasn't yet made any updates)
- MAX_NUMBER_OF_DYNAMIC_RULES  and MAX_NUMBER_OF_SESSION_RULES - new entries, including documentation links (Chrome release number determined from https://chromium.googlesource.com/chromium/src/+/eab7fc99e59b69e929d02e43bdcf8bbd75333869/chrome/VERSION)
- MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES  and MAX_NUMBER_OF_UNSAFE_SESSION_RULES - new entries, excluding documentation links as these haven't been been implemented in Firefox yet (to be implemented in [bug 1894119](https://bugzilla.mozilla.org/show_bug.cgi?id=1894119))

#### Related issues

Related changes to MDN content in https://github.com/mdn/content/pull/34214

### CODE REVIEW

File: webextensions/api/declarativeNetRequest.json [hunk 1]
```diff
@@ -148,7 +148,8 @@
               },
```
Comment (dotproto) [inline webextensions/api/declarativeNetRequest.json:148]: I reached out to @oliverdunk about Chrome's support of the old property. It appears that while Chrome did not formally mark this property as deprecated, the intent was to replace it. The fact that `nodoc` was added to `MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES` in [the same CL](https://source.chromium.org/chromium/chromium/src/+/cbcff7e25340834b9ef92cdb05e20ab4f3a34564) that removed the `nodoc` flags from the new properties is indicative of Chrome's intent to deprecate this property. 

See also WIP Chromium [CL 5637030](https://chromium-review.googlesource.com/c/chromium/src/+/5637030).

```suggestion
                "note": "Deprecated in Chrome 120"
              },
```

**Review by dotproto — CHANGES_REQUESTED** [review]

File: webextensions/api/declarativeNetRequest.json (hunk 1 above)
Comment (dotproto) [reply webextensions/api/declarativeNetRequest.json:149]: Resolved.

**Review by dotproto — APPROVED** [review]

**Comment by queengooborg** [comment]
Hi @rebloor -- this PR should not have been merged yet if there was a failing test, as BCD requires CI tests to pass before merging.

**Comment by rebloor** [comment]
@queengooborg I wasn't aware of the linter behavior regarding the addition of URLs once documentation is published. Where would I have found information about this new behavior? So, from now on, for new docs, I shouldn't include the "mdn_url" but wait for them to be added automatically.

**Comment by queengooborg** [comment]
Sorry for not responding to this earlier, this got buried in my notifications.

New linter behavior isn't something that we've documented in BCD, as it is an internal change that impacts contributors, not consumers.  (Actually, hardly any specific linter rules are documented; only the ones that were originally non-linted policies are documented.) New linter rules are, however, almost always discussed during the BCD weekly calls and aren't implemented before they receive owner consensus.

The motivation for the new MDN URL linter is to reduce the need for manual updates to MDN URLs when pages are created, moved or deleted.  So far, it's been mostly helpful, though this is a case where it hasn't been quite so helpful (the pages were updated but the linter didn't have the updated information yet -- which is why I brought up the idea of making the URL linter warn instead of error).

---

The important part is just making sure that the CI always returns good before merging a PR, even if the issue will eventually fix itself when dependencies are updated.  This is because unlike MDN content, the CI tests for PRs run on EVERY file, not just the ones that are modified.  Thus, if a PR with a failing test is merged, all other open PRs suddenly have failing tests, which halts all BCD development until the CI passes again!

**Comment by rebloor** [comment]
@queengooborg thanks for that, should I come across another similar issue I'll check with you first rather than assuming it's an issue that would be otherwise fixed. (Are there minutes from the weekly call? Something I could search if I do come across the situations.)

**Comment by queengooborg** [comment]
There are meeting notes, yes!  Here they are: https://docs.google.com/document/d/e/2PACX-1vS8QeLHJNMAohmVLDLM0d-6EYo1H5Pw95xz9BFHmkmLWoj80H19XZVJdzYvd2YPig6CaC-8x7lpQm9W/pub

**Pull request merged**